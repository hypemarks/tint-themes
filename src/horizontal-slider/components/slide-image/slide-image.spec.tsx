import { render } from '@testing-library/react';
import React from 'react';
import SlideImage from './slide-image';
import { buildPost } from '../../../../test/generate';
import { IPostWithCTA } from '../../..';

describe('Test <SlideImage /> Component', () => {
  const postWithCTA = buildPost({
    cta: [{ text: 'text', url: 'url', thumbnail_url: null, type: 'cta', coordinates: null, id: 'id' }],
  });

  it('should render snapshot', () => {
    const wrapper = render(
      <SlideImage
        post={postWithCTA as IPostWithCTA}
        config={{
          url: 'https://i.picsum.photos/id/230/200/300.jpg?hmac=pyhlpgJN2oBeEzhJbnJYrCsLoJM6MKd_NUQGIQhVx5k',
        }}
        ctaConfig={{
          ctaButtonColor: null,
          ctaButtonShow: null,
          ctaButtonTextColor: null,
          appendUtmToCta: null,
        }}
        onCtaClick={() => {}}
        onClick={() => {}}
        nextCta={() => {}}
        prevCta={() => {}}
        activeCTAIndex={0}
        shouldDisplayCta={false}
        tabIndex={0}
      />
    );

    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
