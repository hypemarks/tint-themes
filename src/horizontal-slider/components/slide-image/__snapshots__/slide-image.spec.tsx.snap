// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <SlideImage /> Component should render snapshot 1`] = `
<DocumentFragment>
  <div
    class="slide-image"
  />
</DocumentFragment>
`;
